drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-client/src/main/java/org/drools/workbench/screens/scenariosimulation/kogito/client/dmn/AbstractKogitoDMNService.java

```diff
@@ -0,0 +1,469 @@
+                } else {
+                    feelPrimitiveType = new ClientDMNType(URI_FEEL, name, null, false, false, null, type);
+                }
+                toReturn.put(name, feelPrimitiveType);
+            }
+        }
+        jsitItemDefinitions.sort(getItemDefinitionComparator());
+        for (int i = 0; i < jsitItemDefinitions.size(); i++) {
+            final JSITItemDefinition jsitItemDefinition = Js.uncheckedCast(jsitItemDefinitions.get(i));
+            toReturn.put(jsitItemDefinition.getName(), getDMNType(jsitItemDefinition, nameSpace, toReturn));
+        }
+        return toReturn;
+    }
+
+    /**
+     * This comparator sorts a collection of <code>JSITItemDefinition</code> in order to have dependent items, referred by
+     * typeRef field, BEFORE their referred items.
+     * @return
+     */
+    protected Comparator<JSITItemDefinition> getItemDefinitionComparator() {
+        return (o1, o2) -> {
+            if (o1.getTypeRef() == null && o2.getTypeRef() == null) {
+                return 0;
+            }
+            if (o1.getTypeRef() == null) {
+                return -1;
+            }
+            if (o2.getTypeRef() == null) {
+                return 1;
+            }
+            if (o1.getTypeRef().equals(o2.getName())) {
+                return 1;
+            }
+            return 0;
+        };
+    }
+
+    /**
+     * This method creates a <code>ClientDMNType</code> object of a given <code>JSITItemDefinition</code> object.
+     * To correctly work, it requires a sorted <code>dmnTypesMap</code>, in order to process items which don't refer to
+     * a "super items" BEFORE items with a refer to another items.
+     * @param itemDefinition
+     * @param namespace
+     * @param dmnTypesMap
+     * @return
+     */
+    public ClientDMNType getDMNType(final JSITItemDefinition itemDefinition,
+                                    final String namespace,
+                                    final Map<String, ClientDMNType> dmnTypesMap) {
+        final Map<String, ClientDMNType> fields = new HashMap<>();
+        boolean isCollection = itemDefinition.getIsCollection();
+        /* First Step: inheriting fields defined from item's typeRef, which represent its "super item".
+         *  This is required when a typeRef is defined for current itemDefinition */
+        if (itemDefinition.getTypeRef() != null) {
+            final ClientDMNType clientDmnType = dmnTypesMap.get(itemDefinition.getTypeRef());
+            if (clientDmnType != null) {
+                /* Fields are added if the referred "super item" it's a composite (eg. it holds user defined fields) */
+                if (clientDmnType.isComposite()) {
+                    fields.putAll(clientDmnType.getFields());
+                }
+                /* If "super item" is a collection, current item must inherit this property */
+                isCollection = clientDmnType.isCollection() ? clientDmnType.isCollection() : isCollection;
```

Comment:
We can simplify it as `clientDmnType.isCollection() || isCollection`

Reply:
@danielezonca done